### PR TITLE
Finer grained immutable workspace caching

### PR DESCRIPTION
**Pull request by asodja**

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test PT

**Comment by asodja** [comment]
@bot-gradle help

**Comment by asodja** [comment]
@bot-gradle test PT

**Comment by asodja** [comment]
@bot-gradle test PT

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle clean

**Comment by asodja** [comment]
@bot-gradle test APT

**Comment by asodja** [comment]
@bot-gradle test this APT

**Comment by asodja** [comment]
@bot-gradle test this APT

**Comment by asodja** [comment]
@bot-gradle test this

**Review by eskatos — APPROVED** [review]
Looks good from the Kotlin DSL pov.

File: platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/workspace/impl/FinerGrainedImmutableWorkspaceProvider.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    private final ExecutionHistoryStore executionHistoryStore;
+    private final PersistentCache cache;
+    private final Map<String, PersistentCache> finerGrainedCaches;
+    private final Function<String, CacheBuilder> finerCacheFactory;
+
+    private FinerGrainedImmutableWorkspaceProvider(
+        CacheBuilder cacheBuilder,
+        Function<String, CacheBuilder> finerCacheFactory,
+        FileAccessTimeJournal fileAccessTimeJournal,
+        ExecutionHistoryStore historyFactory,
```
Comment (lptr) [inline platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/workspace/impl/FinerGrainedImmutableWorkspaceProvider.java:61]: ❓ Why is this called a factory?

**Comment by asodja** [comment]
We decided we will rather use move sematics and not finer grained caching. See https://github.com/gradle/gradle/pull/26970

**Pull request closed without merging**